### ISSUE TITLE
User request: one click move

### ISSUE THREAD

**Issue by Siderite** [lichess-org/chessground#305]
I have a player who is disabled and can't use their fingers well. To them every click is a burden and drag and drop impossible. Therefore a way to click only once to move a piece is very useful.

  The suggestion is to implement an additional optional mode of  piece movement by clicking on an empty square and, if only one piece can legally move to that square, perform the move.

  I am aware that defining a legal move is outside the purview of chessground, but the mechanism should be there, perhaps with some custom parameters like the Lichess dests array to define possible moves.

**Comment by pawndini** [comment]
this feature would really make life easier for me, and even for people that are just slow in using your mouse, this would definitely speed at things.  I think doing puzzles and a more brisk flow could possibly help you recognize the structure thhe next time you see it.  Also, if there are three different pieces that can go to that square, maybe give the user the option to pick one…?

Anyway, great idea!

**Comment by TBestLittleHelper** [comment]
This is https://github.com/lichess-org/lila/issues/4738

**Comment by gbtami** [comment]
Because of chessground declared itself as "No chess logic inside", this feature completely belongs to lila (not here) IMO.
I mean this library code has no info about valid moves nor about unambiguous moves, it is impossible to implement it here.

**Comment by Siderite** [comment]
chessground already has the concept of dests, right? Which are determined by the software using it, in this case Lila. So the code to determine the one click behavior, given each piece has the destinations defined, should be here.

**Comment by gbtami** [comment]
True. This may work for standard chess, but not for crazyhouse drop moves atm.

**Comment by Siderite** [comment]
I don't know what crazy chess does, but if it doesn't use dests, then it probably doesn't populate it, so you don't enable the feature.

**Comment by Siderite** [comment]
To be honest, I've already implemented in LiChess Tools a hacky version of this for play/puzzles, but I have not enabled it because changing the way people play may be considered an unfair advantage. The version for Analysis works fine, though.

Anyway, the problem is that until Lila doesn't accept to make the effort to implement this, people like the player I mentioned cannot comfortably play chess. If the feature was available as an option in chessground, I am sure they would move on it, though.

**Comment by gbtami** [comment]
All the lichess games are using dests. Here is an example
[rnb1kbnr/ppp2ppp/8/3qp3/8/8/PPPP1PPP/RNBQKBNR/Pp w KQkq - 0 4](https://lichess.org/analysis/crazyhouse/rnb1kbnr/ppp2ppp/8/3qp3/8/8/PPPP1PPP/RNBQKBNR%5BPp%5D_w_KQkq_-_0_4?color=white)
Chessground knows nothing about the pockets atm, so if you enable one click move, clicking on c4 square will move the white bishop there. But it would be possible to drop the pawn there as well. This is the problem I mentioned above.
